### PR TITLE
Example of how StrictMode is not aligned with production

### PR DESCRIPTION
**Pull request by sammy-SC**
Without StrictMode, Offscreen in Suspense does not disconnect passive effects when a component is suspended. It disconnects layout effects but for backward compatibility, passive effects are not disconnected. This is for backwards compatibility. Ideally, we want Offscreen even in Suspense to disconnect passive effects.
StrictMode, however, does disconnect passive effects, as well as layout effects. To unblock sync of React, we will make StrictMode backwards compatible as well.

### CODE REVIEW

File: packages/react-reconciler/src/ReactFiberWorkLoop.old.js [hunk 1]
```diff
@@ -3295,7 +3304,15 @@ function doubleInvokeEffectsInDEVIfNecessary(
     if (isInStrictMode && fiber.flags & Visibility) {
       // Double invoke effects on Offscreen's subtree only
       // if it is visible and its visibility has changed.
-      doubleInvokeEffectsOnFiber(root, fiber);
+
+      // For backwards compatibility, we don't unmount passive effects when a tree suspends.
+      // StrictMode needs to align with this.
+      const includePassiveEffects: boolean =
+        (fiber.stateNode._visibility & OffscreenPassiveEffectsConnected) !==
```
Comment (acdlite) [inline packages/react-reconciler/src/ReactFiberWorkLoop.old.js:3311]: Why is this line necessary? Seems like the tag check below should be sufficient. Regression test passes, too.

**Comment by sammy-SC** [comment]
To clarify, I realise this is not the behaviour we want to have in long term. But, we need it to unblock the sync. Additionally, the check if parent is Suspense is already in the codebase. This PR only makes sure it behaves the same in production and with StrictMode.

File: packages/react-reconciler/src/ReactFiberWorkLoop.old.js (hunk 1 above)
Comment (sammy-SC) [reply packages/react-reconciler/src/ReactFiberWorkLoop.old.js]: Let me remove it. I added it because that's what the actual implementation does.

**Review by acdlite — APPROVED** [review]
Yay great find!

**Comment by acdlite** [comment]
Yeah though I do wonder if we should remove the special case from the Strict Mode behavior even before we remove it from Suspense behavior in production. Then other libraries that currently depend on this would be encouraged to fix it, like how we flushed out this change in Relay. But yeah let’s fix Relay first.

**Comment by sammy-SC** [comment]
I just ran WWW sync and this is not sufficient to fix all problems. I'll have to come back to this and investigate Relay's test to understand what went wrong.
